barretenberg/ts/src/barretenberg/backend.ts

```diff
@@ -145,15 +155,19 @@ export class UltraHonkBackend {
   protected api!: Barretenberg;
   protected acirUncompressedBytecode: Uint8Array;
 
-  constructor(acirBytecode: string, protected options: BackendOptions = { threads: 1 }) {
+  constructor(
+    acirBytecode: string,
+    protected backendOptions: BackendOptions = { threads: 1 },
+    protected circuitOptions: CircuitOptions = { recursive: false },
```

Comment:
I'd like to avoid having the recursive flag being stored on the backend if possible. Ideally we'd pass this in to `generateProof`, etc. as an argument so that we can generate both kinds of proofs.

Reply:
One problem I saw with this was that the [initialize()](https://github.com/AztecProtocol/aztec-packages/blob/599f217264d6ded1f0bffff41de5dc579f93f522/barretenberg/ts/src/barretenberg/backend.ts#L33-L48) method needs the `recursive` flag to get the circuit size and initialise the proving key, which are used to initialise `this.composer` and `this.api` as well. It didn't seem right that you can subsequently use a different parameter to `generateProof`, `verifyProof`, `getVerificationKey`, all of which call `instantiate` but only the first call does anything. I'm not sure why the docs of `instantiate` say "ignore", but it does seem like the choice of whether the circuit should generate recursive or normal proofs has to be made once.